// firebase-react-hooks/_lib/auth/index.js
export function firebaseSignIn(instance, namespace, provider, options) {
    if (options === null || options === void 0 ? void 0 : options.redirect) {
        return instance.signInWithRedirect(getProvider(namespace, provider));
    }
    else {
        return instance.signInWithPopup(getProvider(namespace, provider));
    }
}
export function firebaseSendSignInLink(instance, options) {
    return instance.sendSignInLinkToEmail(options.email, {
        url: options.url,
        handleCodeInApp: true
    });
}
export function firebaseSignInWithLink(instance, options) {
    return instance.signInWithEmailLink(options.email, options.url);
}
export function firebaseSignOut(auth) {
    return auth.signOut();
}
function getProvider(auth, provider) {
    switch (provider) {
        case 'twitter':
            return new auth.TwitterAuthProvider();
        case 'google':
            return new auth.GoogleAuthProvider();
        case 'github':
            return new auth.GithubAuthProvider();
    }
}
